lib/ansible/plugins/action/reboot.py

```diff
@@ -0,0 +1,277 @@
+# (c) 2016-2018, Matt Davis <mdavis@ansible.com>
+# (c) 2018, Sam Doran <sdoran@redhat.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import random
+import time
+
+from datetime import datetime, timedelta
+
+from ansible.errors import AnsibleError
+from ansible.plugins.action import ActionBase
+from ansible.module_utils._text import to_native, to_text
+
+
+try:
+    from __main__ import display
+except ImportError:
+    from ansible.utils.display import Display
+    display = Display()
+
+
+class TimedOutException(Exception):
+    pass
+
+
+class ActionModule(ActionBase):
+    TRANSFERS_FILES = False
+
+    DEFAULT_REBOOT_TIMEOUT = 600
+    DEFAULT_CONNECT_TIMEOUT = None
+    DEFAULT_PRE_REBOOT_DELAY = 0
+    DEFAULT_POST_REBOOT_DELAY = 0
+    DEFAULT_TEST_COMMAND = 'whoami'
+    DEFAULT_UPTIME_COMMAND = 'who -b'
+    DEFAULT_REBOOT_MESSAGE = 'Reboot initiated by Ansible'
+    DEFAULT_SHUTDOWN_COMMAND = 'shutdown'
+    DEFAULT_SUDOABLE = True
+
+    SHUTDOWN_COMMANDS = {
+        'linux': DEFAULT_SHUTDOWN_COMMAND,
+        'freebsd': DEFAULT_SHUTDOWN_COMMAND,
+        'sunos': '/usr/sbin/shutdown',
+        'darwin': '/sbin/shutdown',
+    }
+
+    SHUTDOWN_COMMAND_ARGS = {
+        'linux': '-r {delay_min} "{message}"',
+        'freebsd': '-r +{delay_sec}s "{message}"',
+        'sunos': '-y -g {delay_sec} -r "{message}"',
+        'darwin': '-r +{delay_min_macos} "{message}"'
+    }
+
+    def deprecated_args(self):
+        deprecated_args = self._task.args.get('DEFAULT_DEPRECATED_ARGS')
```

Comment:
shouldn't this line and the one below just not be here, and the loop be `for arg, version in self.DEFAULT_DEPRECATED_ARGS` (though those aren't really a default either, so `DEFAULT` is a bit of a misnomer)

Reply:
Yup.

Reply:
I made the changes you suggested.